### PR TITLE
iio: Adapt iio_axi_dac, iio_axi_adc to new iio. Update projects

### PR DESCRIPTION
**Pull request by cristiantpop**
In this moment we have a new optimised iio version and an old one. Adapt
all projects to new iio and remove the old one.

Signed-off-by: Cristian Pop <cristian.pop@analog.com>

### CODE REVIEW

File: iio/iio_axi_adc/iio_axi_adc.c [hunk 1]
```diff
@@ -659,7 +448,101 @@ static ssize_t iio_axi_adc_read_dev(void *iio_inst, char *pbuf, size_t offset,
...
+	uint16_t i = 0;
+
+	if (!iio_device)
+		return FAILURE;
+
+	if (iio_device->channels) {
+		while (iio_device->channels[i]) {
+			if (iio_device->channels[i]->name)
+				free(iio_device->channels[i]->name);
+			if (iio_device->channels[i])
```
Comment (mchindri) [inline iio/iio_axi_adc/iio_axi_adc.c:467]: The check is redundatnt. It is also checked in `while (iio_device->channels[i]) `

File: iio/iio_axi_dac/iio_axi_dac.c [hunk 2]
```diff
@@ -591,334 +574,270 @@ enum ch_type {
...
+//	iio_device->channels[i] = NULL;
+//
+//	return iio_device;
+//
+//error:
+//	iio_axi_dac_delete_device(iio_device);
+//
+//	return NULL;
+//}
+//
```
Comment (mchindri) [inline iio/iio_axi_dac/iio_axi_dac.c:672]: Remove lines

**Review by mchindri — COMMENTED** [review]
In the final version the changes should be splitted in multiple commits

File: iio/iio_ad713x/iio_ad713x.c [hunk 3]
```diff
@@ -204,79 +139,47 @@ ssize_t iio_ad713x_read_dev(void *iio_inst, char *pbuf, size_t offset,
...
-		.transfer_dev_to_mem = iio_ad713x_transfer_dev_to_mem,
-		.read_data = iio_ad713x_read_dev,
-		.transfer_mem_to_dev = NULL,
-		.write_data = NULL,
-	};
-
-	status = iio_register(iio_interface);
-	if (status < 0)
-		goto error_free_iio_interface;
+	iio_ad713x->dev_descriptor.num_ch = param->num_channels;
```
[the remark this answers is not in the document]
Comment (cristiantpop) [reply iio/iio_ad713x/iio_ad713x.c:171]: I am not sure what are you saying, but it seems to me that this pattern of struct is present in the other projects. Why not stick to it?

**Comment by cristiantpop** [comment]
V2:

- Update remaining projects

- Remove comments

- Reuse variables

- Remove redundant checks

- projects/ad713x_fmcz/src.mk: check if libiio needs to be included

**Comment by cristiantpop** [comment]
V3: 

- Fix conflict in linux.mk

- Fix doxygen

File: iio/iio_axi_dac/iio_axi_dac.c [hunk 4]
```diff
@@ -591,334 +574,173 @@ enum ch_type {
...
 		return FAILURE;
 
-	if(iio_device->channels) {
+	if (iio_device->channels) {
 		while (iio_device->channels[i]) {
-			if(iio_device->channels[i]->name)
+			if (iio_device->channels[i]->name)
 				free(iio_device->channels[i]->name);
-			if(iio_device->channels[i])
+			if (iio_device->channels[i])
```
Comment (mchindri) [inline iio/iio_axi_dac/iio_axi_dac.c:594]: Here is also redundant the check

**Review by mchindri — COMMENTED** [review]
Also, it seems that doxygen complains were not fully fixed

**Comment by cristiantpop** [comment]
V4:

- Add some comments

**Comment by cristiantpop** [comment]
V5:

- Split in 3 commits

**Comment by cristiantpop** [comment]
V6:
- Rename iio_axi channels to "voltage"/"altvoltage"

**Comment by cristiantpop** [comment]
V7:

- Fix astyle

**Review by mchindri — APPROVED** [review]
LGTM

**Review by dbogdan — APPROVED** [review]

**Pull request merged**